src/__tests__/Parser.test.ts

```diff
@@ -126,23 +140,30 @@ test('parse nested group properly', () => {
     new Multiplication(
       new Group(
         new Addition(
-          new Group(
-            new Addition(
-              new Num(5),
-              new Num(4),
-            ),
-          ),
-          new Num(1),
-        ),
+          new Group(new Addition(new Num(5), new Num(4))),
+          new Num(1)
+        )
       ),
-      new Num(3),
-    ),
+      new Num(3)
+    )
   )
 })
 
-
 test('parse throws an error if there are unparsed tokens', () => {
   expect(() => {
-    parse(scan('(3 + 4) 4'))
+    parse(scan('(3 + 4) 5'))
   }).toThrow(/.*unparsed tokens.*/)
 })
+
+test('is addition also valid multiplaction?', () => {
```

Comment:
this should throw an error - can you instead wrap this in an expect.toThrow? ie:

```ts
expect(() => {
   let parser = new Parser(scan('5 + 4'))
   parser.multiplication()
}).toThrow(...)

```